packages/itwinui-react/src/core/Table/TableRowMemoized.tsx

```diff
@@ -223,7 +223,5 @@ export const TableRowMemoized = React.memo(
       nextProp.state.sticky.isScrolledToLeft &&
     prevProp.state.sticky.isScrolledToRight ===
       nextProp.state.sticky.isScrolledToRight &&
-    prevProp.density === nextProp.density &&
-    prevProp.virtualItem === nextProp.virtualItem &&
```

Comment:
I think this is a bit problematic. We are accepting `virtualItem` as a prop but we aren't using it during memoization, which could result in a stale version of this prop being used.

Reply:
I found that we were only using the `vitualItem.index` and `virtualItem.start` values in the `TableRow`, so I tried memoizing on those values specifically. I found that memoizing on these values does not cause unnecessary rerenders and does not negatively impact performance. I added these values to the memoization in this [commit](https://github.com/iTwin/iTwinUI/pull/2092/commits/d4b2e3cb4734afdb4225da19ed1e03c650f5ab6d).